tests/snappi_tests/intf_utils/intf_accept_metrics.py

```diff
@@ -0,0 +1,147 @@
+# This file defines the interfaces that snappi tests accept external metrics.
+
+#  Metrics data are organized into the hierarchies below
+#  TestMetrics
+#    ├── TestID
+#    └── DeviceMetrics
+#       ├── DeviceID
+#       └── Metric
```

Comment:
create a generic Metric class that represents a single metric, which contains:
- description/labels: Name, Description, unit, ....
- Value: single layer is good enough with inheritance.
- Reporter: Reference to MetricsReporter. Register itself to Reporter when created, so Reporter can gather all metrics after everything is changed.

```python
class Metric...:
    def __init__(name, ...., reporter):
        reporter.add_metric(self)
        ....

class GaugeMetric:
....


reporter = MetricReporterFactory(...).build()
port_rx = GaugeMetric(...., reporter)

port_rx.set(123)
reporter.report(time)
```

Reply:
Hence, ultimately the final code for people to use would be:

```
metrics = {
"PortRx" = GaugeMetric(......, reporter)
....
}

for r in csv:
    for c in r:
        metric[c.title].set(c.value)

reporter.report(time)
```